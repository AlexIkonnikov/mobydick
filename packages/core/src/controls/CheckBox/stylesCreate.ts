import {StyleSheet} from 'react-native';

import {IThemeContext} from '../../styles/types';
import rem from '../../styles/utils/rem';

const stylesCreate = (
  themeColors: IThemeContext,
  disabled = false,
  selected = false,
  width = rem(20),
  height = rem(20),
) => {
  const {colors, spaces} = themeColors;

  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: 'transparent',
      opacity: disabled ? 0.4 : 1,
      maxWidth: '100%',
    },
    checkbox: {
      borderWidth: selected ? 0 : spaces.Space2,
      borderColor: colors.BorderNormal,
      width: width,
      height: height,
      borderRadius: spaces.Space4,
    },
  });
};

export default stylesCreate;
