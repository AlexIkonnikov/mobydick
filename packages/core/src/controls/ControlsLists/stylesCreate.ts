import {StyleSheet} from 'react-native';

import {IThemeContext} from '../../styles/theme/types';

const stylesCreate = (_theme: IThemeContext, horizontal: boolean) =>
  StyleSheet.create({
    list: {
      flexDirection: horizontal ? 'row' : 'column',
    },
  });

export default stylesCreate;
