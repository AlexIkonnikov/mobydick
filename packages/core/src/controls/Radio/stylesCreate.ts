import rem from '../../styles/utils/rem';
import {createStyles} from '../../styles';

const stylesCreate = createStyles(
  ({spaces, colors}, selected: boolean, disabled: boolean) => ({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: 'transparent',
      opacity: disabled ? 0.4 : 1,
    },
    circle: {
      width: rem(22),
      aspectRatio: 1,
      borderRadius: rem(11),
      backgroundColor: selected ? colors.ElementBase : 'transparent',
      borderColor: selected ? colors.ElementBase : colors.BorderNormal,
      borderWidth: spaces.Space2,
      justifyContent: 'center',
      alignItems: 'center',
    },
    innerCircle: {
      width: rem(18),
      aspectRatio: 1,
      borderWidth: spaces.Space2,
      borderRadius: rem(9),
      borderColor: colors.BgPrimary,
    },
  }),
);

export default stylesCreate;
