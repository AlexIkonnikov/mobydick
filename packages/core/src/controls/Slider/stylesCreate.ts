import {I18nManager, StyleSheet} from 'react-native';

import {IThemeContext} from '../../styles';

const stylesCreate = ({colors, spaces}: IThemeContext) => {
  return StyleSheet.create({
    controlsContainer: {
      flexDirection: 'row',
      justifyContent: I18nManager.isRTL ? 'flex-end' : 'flex-start',
      alignItems: 'center',
      minHeight: spaces.Space32,
    },
    rail: {
      flex: 1,
      height: spaces.Space4,
      borderRadius: spaces.Space2,
      backgroundColor: colors.IconMuted,
    },
    selectedRailContainer: {
      position: 'absolute',
    },
    selectedRail: {
      height: spaces.Space4,
      backgroundColor: colors.IconBase,
      borderRadius: spaces.Space2,
    },
    highThumbContainer: {
      position: 'absolute',
    },
    railsContainer: {
      ...StyleSheet.absoluteFillObject,
      flexDirection: 'row',
      alignItems: 'center',
    },
    touchableArea: {
      ...StyleSheet.absoluteFillObject,
    },
  });
};

export default stylesCreate;
