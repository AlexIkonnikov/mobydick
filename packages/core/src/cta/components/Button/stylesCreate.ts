import {StyleSheet, ViewStyle} from 'react-native';

import rem from '../../../styles/utils/rem';
import {IThemeContext} from '../../../styles/types';

import {IButtonSize, IButtonTypes} from './types';

interface IButtonContents {
  theme: IThemeContext;
  size: IButtonSize;
  leftIcon: boolean;
  rightIcon: boolean;
  text: boolean;
}

const getButtonHorizontalPadding = ({
  theme,
  leftIcon,
  rightIcon,
  text,
}: Omit<IButtonContents, 'size'>) => {
  if (!text) {
    return theme.spaces.Space6;
  }
  if (leftIcon || rightIcon) {
    return rem(14);
  }
  return theme.spaces.Space12;
};

const getButtonStyles = ({
  theme,
  size,
  leftIcon,
  rightIcon,
  text,
}: IButtonContents): ViewStyle => {
  switch (size) {
    case IButtonSize.small:
      return {
        minHeight: theme.spaces.Space32,
        paddingVertical: theme.spaces.Space6,
        paddingHorizontal: getButtonHorizontalPadding({
          theme,
          leftIcon,
          rightIcon,
          text,
        }),
      };
    case IButtonSize.large:
      return {
        minHeight: theme.spaces.Space48,
        paddingVertical: theme.spaces.Space12,
        paddingHorizontal: text ? rem(26) : theme.spaces.Space12,
      };
    case IButtonSize.fixed:
      return {
        alignSelf: 'stretch',
        minHeight: theme.spaces.Space48,
      };
    default:
      return {};
  }
};

const primaryStyle = ({
  theme,
  size,
  leftIcon,
  rightIcon,
  text,
}: IButtonContents) => {
  const {colors} = theme;

  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: colors.CtaBtnPrimary,
      borderRadius: theme.spaces.Space12,

      ...getButtonStyles({theme, size, leftIcon, rightIcon, text}),
    },

    text: {
      paddingHorizontal:
        size === IButtonSize.small ? theme.spaces.Space4 : theme.spaces.Space6,
    },
    counter: {
      marginLeft: theme.spaces.Space2,
    },
  });
};

const secondaryStyle = ({
  theme,
  size,
  leftIcon,
  rightIcon,
  text,
}: IButtonContents) => {
  const defaultStyles = primaryStyle({theme, size, leftIcon, rightIcon, text});
  const {
    colors: {CtaBtnSecondary},
  } = theme;

  defaultStyles.container.backgroundColor = CtaBtnSecondary;

  return defaultStyles;
};

const tertiaryStyle = ({
  theme,
  size,
  leftIcon,
  rightIcon,
  text,
}: IButtonContents) => {
  const defaultStyles = primaryStyle({theme, size, leftIcon, rightIcon, text});

  defaultStyles.container.backgroundColor = 'transparent';

  return defaultStyles;
};

const disabledStyle = ({
  theme,
  size,
  leftIcon,
  rightIcon,
  text,
}: IButtonContents) => {
  const defaultStyles = primaryStyle({theme, size, leftIcon, rightIcon, text});
  const {
    colors: {CtaBtnMuted},
  } = theme;

  defaultStyles.container.backgroundColor = CtaBtnMuted;

  return defaultStyles;
};

const destructiveStyle = ({
  theme,
  size,
  leftIcon,
  rightIcon,
  text,
}: IButtonContents) => {
  const defaultStyles = primaryStyle({theme, size, leftIcon, rightIcon, text});
  const {
    colors: {CtaBtnDestructive},
  } = theme;

  defaultStyles.container.backgroundColor = CtaBtnDestructive;

  return defaultStyles;
};

const stylesCreate = (
  theme: IThemeContext,
  type: IButtonTypes,
  size: IButtonSize,
  leftIcon: boolean,
  rightIcon: boolean,
  text: boolean,
) => {
  switch (type) {
    case IButtonTypes.secondary:
      return secondaryStyle({theme, size, leftIcon, rightIcon, text});
    case IButtonTypes.tertiary:
      return tertiaryStyle({theme, size, leftIcon, rightIcon, text});
    case IButtonTypes.disabled:
      return disabledStyle({theme, size, leftIcon, rightIcon, text});
    case IButtonTypes.destructive:
      return destructiveStyle({theme, size, leftIcon, rightIcon, text});
    case IButtonTypes.primary:
    default:
      return primaryStyle({theme, size, leftIcon, rightIcon, text});
  }
};
export default stylesCreate;
