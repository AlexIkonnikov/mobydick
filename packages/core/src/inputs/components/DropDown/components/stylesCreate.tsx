import {createStyles} from '../../../../styles';

const stylesCreate = createStyles(({spaces, colors}) => ({
  flatList: {
    position: 'absolute',
    backgroundColor: colors.BgSecondary,
    borderRadius: spaces.Space12,
    borderWidth: spaces.Space1,
    borderColor: colors.BorderSoft,
    paddingVertical: spaces.Space8,
  },

  dropDownItem: {
    justifyContent: 'center',
    paddingHorizontal: spaces.Space12,
    paddingVertical: spaces.Space8,
  },
}));

export default stylesCreate;
