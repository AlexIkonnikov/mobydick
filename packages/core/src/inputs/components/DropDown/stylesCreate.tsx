import {StyleSheet} from 'react-native';

import {IInputsTypes} from '../types';
import {disabledStyle, validStyle, wrongStyle} from '../../style';
import rem from '../../../styles/utils/rem';
import {IThemeContext} from '../../../styles/types';

const defaultStyle = (theme: IThemeContext) => {
  const {colors, spaces} = theme;
  return StyleSheet.create({
    container: {
      justifyContent: 'center',
    },
    inputContainer: {
      backgroundColor: colors.BgSecondary,
      borderRadius: spaces.Space8,
      paddingLeft: rem(18),
      paddingRight: theme.spaces.Space12,
      borderWidth: spaces.Space1,
      borderColor: colors.BorderNormal,
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      marginBottom: spaces.Space8,
    },
    pv8: {
      paddingVertical: spaces.Space8,
    },
    placeholder: {
      flex: 1,
    },
    title: {
      paddingBottom: spaces.Space8,
    },
    overlay: {
      position: 'absolute',
      width: '100%',
      height: '100%',
      backgroundColor: colors.TextSecondary,
    },
  });
};

const stylesCreate = (
  theme: IThemeContext,
  type: IInputsTypes,
  isOpen: boolean,
) => {
  switch (type) {
    case IInputsTypes.valid:
      return validStyle(theme, defaultStyle(theme), isOpen);
    case IInputsTypes.wrong:
      return wrongStyle(theme, defaultStyle(theme), isOpen);
    case IInputsTypes.disabled:
      return disabledStyle(theme, defaultStyle(theme), isOpen);
    case IInputsTypes.default:
    default:
      return defaultStyle(theme);
  }
};

export default stylesCreate;
