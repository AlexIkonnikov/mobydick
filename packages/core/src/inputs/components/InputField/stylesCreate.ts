import {StyleSheet} from 'react-native';

import {IInputsTypes} from '../types';
import {disabledStyle, validStyle, wrongStyle} from '../../style';
import {IThemeContext, rem} from '../../../styles';

const defaultStyle = (
  theme: IThemeContext,
  focused: boolean,
  multiline: boolean,
) => {
  const {colors, spaces} = theme;
  return StyleSheet.create({
    container: {
      minWidth: rem(130),
    },
    inputContainer: {
      backgroundColor: colors.BgSecondary,
      paddingRight: spaces.Space16,
      borderRadius: spaces.Space8,
      borderWidth: spaces.Space1,
      borderColor: focused ? colors.BorderNormal : 'transparent',
      marginVertical: spaces.Space8,
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: multiline ? undefined : 'center',
    },
    textInput: {
      flex: 1,
      paddingLeft: rem(18),
      textAlignVertical: multiline ? 'top' : undefined,
      padding: 0, // Android по дефолту ставит padding на input's
      paddingVertical: multiline ? spaces.Space12 : 0,
      borderRadius: spaces.Space8,
      // backgroundColor: 'red',
    },

    androidTextInput: {
      fontSize: spaces.Space16,
      color: colors.TextPrimary,
      padding: 0, // Android по дефолту ставит padding на input's
    },
  });
};

const stylesCreate = (
  theme: IThemeContext,
  type: IInputsTypes,
  focused: boolean,
  multiline: boolean,
) => {
  switch (type) {
    case IInputsTypes.valid:
      return validStyle(
        theme,
        defaultStyle(theme, focused, multiline),
        focused,
      );
    case IInputsTypes.wrong:
      return wrongStyle(
        theme,
        defaultStyle(theme, focused, multiline),
        focused,
      );
    case IInputsTypes.disabled:
      return disabledStyle(
        theme,
        defaultStyle(theme, focused, multiline),
        focused,
      );
    case IInputsTypes.default:
    default:
      return defaultStyle(theme, focused, multiline);
  }
};
export default stylesCreate;
