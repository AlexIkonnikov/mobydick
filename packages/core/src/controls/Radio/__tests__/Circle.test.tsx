import React from 'react';
import {render} from '@testing-library/react-native';
import {renderHook} from '@testing-library/react-hooks';

import Circle from '../Circle';
import stylesCreate from '../stylesCreate';
import useStyles from '../../../styles/theme/hooks/useStyles';

describe('@lad-tech/mobydick-core/Radio/Circle', () => {
  it('renders correctly', () => {
    const {result} = renderHook(() => useStyles(stylesCreate, false, false));
    const [styles] = result.current;
    const {toJSON} = render(
      <Circle innerStyle={styles.circle} outerStyle={styles.innerCircle} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly selected', () => {
    const {result} = renderHook(() => useStyles(stylesCreate, false, false));
    const [styles] = result.current;
    const {toJSON} = render(
      <Circle
        innerStyle={styles.circle}
        outerStyle={styles.innerCircle}
        selected
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
