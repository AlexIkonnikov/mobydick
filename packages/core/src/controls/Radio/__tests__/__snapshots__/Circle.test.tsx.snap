// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-core/Radio/Circle renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "aspectRatio": 1,
      "borderColor": "#FFF",
      "borderRadius": 13.5,
      "borderWidth": 3,
      "width": 27,
    }
  }
/>
`;

exports[`@npm/mobydick-core/Radio/Circle renders correctly selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "aspectRatio": 1,
      "borderColor": "#FFF",
      "borderRadius": 13.5,
      "borderWidth": 3,
      "width": 27,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "transparent",
        "borderColor": "rgba(32, 36, 45, 0.3)",
        "borderRadius": 16.5,
        "borderWidth": 3,
        "justifyContent": "center",
        "width": 33,
      }
    }
  />
</View>
`;
